### PR TITLE
Fix example_emr_serverless system test

### PR DESCRIPTION
**Pull request by vincbeck**
Fix `example_emr_serverless` system test. EMR does not allow cross region calls. The region was hardcoded for a S3 file to `us-east-1`. Any execution of this system test outside of `us-east-1` results in a failure.

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def check_for_bucket(self, bucket_name: str | None = None) -> bool:
             # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.head_bucket
             return_code = int(e.response['Error']['Code'])
             if return_code == 404:
-                self.log.error('Bucket "%s" does not exist', bucket_name)
+                self.log.info('Bucket "%s" does not exist', bucket_name)
```
Comment (eladkal) [inline airflow/providers/amazon/aws/hooks/s3.py:223]: why do we change log level only for 404?

Comment (vincbeck) [reply airflow/providers/amazon/aws/hooks/s3.py:223]: Well. If it's 403 the bucket exists but the user does not have permissions so I think it makes sense to log an error for that use case

Comment (vincbeck) [reply airflow/providers/amazon/aws/hooks/s3.py:223]: On the other side, 404 is a very valid use case. If the bucket exists (and user has permissions), we should not log an error for that

**Review by o-nikolas — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/s3.py (hunk 1 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/hooks/s3.py:223]: @eladkal WDYT?

**Review by eladkal — APPROVED** [review]

**Pull request merged**